src/hotspot/cpu/aarch64/gc/shared/barrierSetAssembler_aarch64.cpp

```diff
@@ -254,7 +254,7 @@ void BarrierSetAssembler::nmethod_entry_barrier(MacroAssembler* masm) {
   __ cmpw(rscratch1, rscratch2);
   __ br(Assembler::EQ, skip);
 
-  __ mov(rscratch1, StubRoutines::aarch64::method_entry_barrier());
+  __ movptr(rscratch1, (uintptr_t) StubRoutines::aarch64::method_entry_barrier());
```

Comment:
Looking how the rest of AArch64 code calls into external things, this might be:

```
 __ lea(rscratch1, ExternalAddress(StubRoutines::aarch64::method_entry_barrier()));
```

...which would end up doing the same (right) thing?

Reply:
lea calls Address::lea, which itself might call movptr, or but it can also construct the address through other means. It won't, not in the way it is currently written. However, movptr() should be call when generating patchable sequences. While this address isn't patched, we do rely on the property that patchable sequences are of a fixed length. The mistake I made before was in not ensuring that property.

Reply:
> lea calls Address::lea, which itself might call movptr, or but it can also construct the address through other means. 

True. LEA is a macro that generates the address in some unspecified way.

Reply:
All right!